diff -Naur Python-3.9.2.orig/Modules/posixmodule.c Python-3.9.2/Modules/posixmodule.c
--- Python-3.9.2.orig/Modules/posixmodule.c	2021-03-20 12:28:02.000000000 +0100
+++ Python-3.9.2/Modules/posixmodule.c	2021-03-20 12:29:00.000000000 +0100
@@ -326,7 +326,7 @@
 #    define HAVE_KILL       1
 #    define HAVE_OPENDIR    1
 #    define HAVE_PIPE       1
-#    define HAVE_SYSTEM     1
+#    undef HAVE_SYSTEM
 #    define HAVE_WAIT       1
 #    define HAVE_TTYNAME    1
 #  endif  /* _MSC_VER */
