diff -Naur numpy-1.24.2.orig/numpy/linalg/setup.py numpy-1.24.2/numpy/linalg/setup.py
--- numpy-1.24.2.orig/numpy/linalg/setup.py     2023-03-11 19:31:34
+++ numpy-1.24.2/numpy/linalg/setup.py  2023-03-11 19:32:12
@@ -78,7 +78,7 @@
     # umath_linalg module
     config.add_extension(
         '_umath_linalg',
-        sources=['umath_linalg.cpp', get_lapack_lite_sources],
+        sources=['umath_linalg.cpp'],
         depends=['lapack_lite/f2c.h'],
         extra_info=lapack_info,
         extra_cxx_compile_args=NPY_CXX_FLAGS,
diff -Naur numpy-1.24.2.orig/numpy/random/setup.py numpy-1.24.2/numpy/random/setup.py
--- numpy-1.24.2.orig/numpy/random/setup.py     2023-03-11 19:38:04
+++ numpy-1.24.2/numpy/random/setup.py  2023-03-11 19:39:48
@@ -139,7 +139,6 @@
     config.add_extension('mtrand',
                          sources=['mtrand.c',
                                   'src/legacy/legacy-distributions.c',
-                                  'src/distributions/distributions.c',
                                  ],
                          include_dirs=['.', 'src', 'src/legacy'],
                          libraries=mtrand_libs,