--- matplotlib-3.7.2.orig/src/tri/_tri.h	2023-07-05 13:43:01
+++ matplotlib-3.7.2/src/tri/_tri.h	2023-09-10 21:43:47
@@ -74,6 +74,7 @@
 
 namespace py = pybind11;
 
+namespace tri {
 
 /* An edge of a triangle consisting of an triangle index in the range 0 to
  * ntri-1 and an edge index in the range 0 to 2.  Edge i goes from the
@@ -799,4 +800,6 @@
     Node* _tree;    // Root node of the trapezoid map search tree.  Owned.
 };
 
+}
+
 #endif
