Index: Python-2.7.1/Modules/posixmodule.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Python-2.7.1/Modules/posixmodule.c	(date 1290868829000)
+++ Python-2.7.1/Modules/posixmodule.c	(revision )
@@ -2644,7 +2644,7 @@
 #endif
 }
 
-
+#undef HAVE_SYSTEM
 #ifdef HAVE_SYSTEM
 PyDoc_STRVAR(posix_system__doc__,
 "system(command) -> exit_status\n\n\
