--- Pillow-6.1.0/setup.py	2014-12-31 20:42:56.000000000 +0530
+++ Pillow-6.1.0/setup.py	2015-02-26 19:38:59.000000000 +0530
@@ -23,3 +25,3 @@
 # monkey patch import hook. Even though flake8 says it's not used, it is.
 # comment this out to disable multi threaded builds.
-import mp_compile
+# import mp_compile
@@ -329,1 +329,2 @@
-            self.parallel = mp_compile.MAX_PROCS
+            # self.parallel = mp_compile.MAX_PROCS
+            pass

