--- materialyoucolor-pyhton/setup.py	2024-03-16 20:29:35.548962347 +0530
+++ materialyoucolor-pyhton.patched/setup.py	2024-03-16 20:31:14.222400319 +0530
@@ -50,6 +50,8 @@
 from functools import lru_cache
 from pathlib import Path
 import urllib.request
+import ssl
+import certifi
 from glob import glob
 from setuptools import find_packages, setup
 from typing import (
@@ -537,7 +539,9 @@
     print("Downloading required files...")
     for file in FILES:
         with open(os.path.join(FOLDER, os.path.basename(file)), "w") as write_buffer:
-            write_buffer.write(urllib.request.urlopen(URL + file).read().decode("utf-8"))
+            write_buffer.write(urllib.request.urlopen(
+                URL + file,  context=ssl.create_default_context(cafile=certifi.where())
+            ).read().decode("utf-8"))
             write_buffer.close()
             print("[Downloaded] : " + file)
 
