diff -Naur numpy/core/setup.py numpy-1.16.4/numpy/core/setup.py
--- numpy-1.16.4.orig/numpy/core/setup.py	2019-05-27 12:41:06.000000000 +0200
+++ numpy-1.16.4/numpy/core/setup.py	2019-09-28 12:51:56.000000000 +0200
@@ -757,7 +757,9 @@
             join('src', 'common', 'numpyos.c'),
             ]
 
-    blas_info = get_info('blas_opt', 0)
+    # XXX IOS, no blas available
+    # blas_info = get_info('blas_opt', 0)
+    blas_info = None
     if blas_info and ('HAVE_CBLAS', None) in blas_info.get('define_macros', []):
         extra_info = blas_info
         # These files are also in MANIFEST.in so that they are always in

--- numpy-1.16.4.orig/numpy/linalg/setup.py	2019-02-21 17:33:42.000000000 +0100
+++ numpy-1.16.4/numpy/linalg/setup.py	2019-09-28 17:34:15.000000000 +0200
@@ -38,9 +38,12 @@
                 return []
             return [all_sources[0]]
 
+    def get_lapack_lite_sources_ios(ext, build_dir):
+        return []
+
     config.add_extension(
         'lapack_lite',
-        sources=['lapack_litemodule.c', get_lapack_lite_sources],
+        sources=['lapack_litemodule.c', get_lapack_lite_sources_ios],
         depends=['lapack_lite/f2c.h'],
         extra_info=lapack_info,
     )
@@ -48,7 +51,7 @@
     # umath_linalg module
     config.add_extension(
         '_umath_linalg',
-        sources=['umath_linalg.c.src', get_lapack_lite_sources],
+        sources=['umath_linalg.c.src', get_lapack_lite_sources_ios],
         depends=['lapack_lite/f2c.h'],
         extra_info=lapack_info,
         libraries=['npymath'],

--- numpy-1.16.4.orig/numpy/core/setup.py	2019-09-28 17:41:37.000000000 +0200
+++ numpy-1.16.4/numpy/core/setup.py	2019-09-28 17:41:58.000000000 +0200
@@ -824,7 +824,10 @@
             join('include', 'numpy', 'npy_1_7_deprecated_api.h'),
             # add library sources as distuils does not consider libraries
             # dependencies
-            ] + npysort_sources + npymath_sources
+
+            # XXX This breaks for iOS, it results on duplicate symbols
+            # ] + npysort_sources + npymath_sources
+            ] #+ npysort_sources + npymath_sources
 
     multiarray_src = [
             join('src', 'multiarray', 'alloc.c'),

--- numpy-1.16.4.orig/numpy/core/setup.py	2019-09-28 17:42:38.000000000 +0200
+++ numpy-1.16.4/numpy/core/setup.py	2019-09-28 17:44:09.000000000 +0200
@@ -926,7 +926,7 @@
 
     config.add_extension('_multiarray_umath',
                          sources=multiarray_src + umath_src +
-                                 npymath_sources + common_src +
+                                 common_src +
                                  [generate_config_h,
                                   generate_numpyconfig_h,
                                   generate_numpy_api,
