--- Python-2.7.13/configure.orig	2017-02-08 10:11:52.000000000 +0000
+++ Python-2.7.13/configure		2017-02-08 10:15:10.000000000 +0000
@@ -3242,6 +3242,9 @@
 	*-*-linux*)
 		ac_sys_system=Linux
 		;;
+	*-*-darwin*)
+		ac_sys_system=Darwin
+		;;
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
@@ -3288,6 +3291,15 @@
 			_host_cpu=$host_cpu
 		esac
 		;;
+	*-*-darwin*)
+		case "$host_cpu" in
+		arm*)
+			_host_cpu=arm
+			;;
+		*)
+			_host_cpu=$host_cpu
+		esac
+		;;
 	*-*-cygwin*)
 		_host_cpu=
 		;;
