diff -Naur Python-3.8.2.orig/Modules/_ctypes/cfield.c Python-3.8.2/Modules/_ctypes/cfield.c
--- Python-3.8.2.orig/Modules/_ctypes/cfield.c	2020-04-13 12:23:46.000000000 +0200
+++ Python-3.8.2/Modules/_ctypes/cfield.c	2020-04-13 12:24:45.000000000 +0200
@@ -1636,7 +1636,7 @@
     struct _ffi_type **elements;
 } ffi_type;
 */
-
+#if 0
 /* align and size are bogus for void, but they must not be zero */
 ffi_type ffi_type_void = { 1, 1, FFI_TYPE_VOID };
 
@@ -1663,5 +1663,5 @@
                                  FFI_TYPE_LONGDOUBLE };
 
 ffi_type ffi_type_pointer = { sizeof(void *), VOID_P_ALIGN, FFI_TYPE_POINTER };
-
+#endif
 /*---------------- EOF ----------------*/
