diff -Naur Pillow-8.2.0.orig/setup.py Pillow-8.2.0/setup.py
--- Pillow-8.2.0.orig/setup.py	2021-04-05 11:11:26.000000000 +0200
+++ Pillow-8.2.0/setup.py	2021-04-05 11:16:12.000000000 +0200
@@ -222,12 +222,9 @@
 
 
 def _find_library_file(self, library):
-    ret = self.compiler.find_library_file(self.compiler.library_dirs, library)
-    if ret:
-        _dbg("Found library %s at %s", (library, ret))
-    else:
-        _dbg("Couldn't find library %s in %s", (library, self.compiler.library_dirs))
-    return ret
+    # The provided find method is broken on our implementation.
+    # We will select by --disable flags what is actually disabled.
+    return True
 
 
 def _find_include_dir(self, dirname, include):
