--- Pillow-11.0.0.orig/setup.py	2024-10-15 01:55:00.000000000 -0400
+++ Pillow-11.0.0/setup.py	2024-11-19 12:20:09.514795844 -0500
@@ -231,12 +231,9 @@
 
 
 def _find_library_file(self: pil_build_ext, library: str) -> str | None:
-    ret = self.compiler.find_library_file(self.compiler.library_dirs, library)
-    if ret:
-        _dbg("Found library %s at %s", (library, ret))
-    else:
-        _dbg("Couldn't find library %s in %s", (library, self.compiler.library_dirs))
-    return ret
+    # The provided find method is broken on our implementation.
+    # We will select by --disable flags what is actually disabled.
+    return True
 
 
 def _find_include_dir(self: pil_build_ext, dirname: str, include: str) -> bool | str:
