diff -Nru libffi-3.2.1/generate-darwin-source-and-headers.py libffi-3.2.1-new/generate-darwin-source-and-headers.py
--- libffi-3.2.1/generate-darwin-source-and-headers.py	2014-11-08 13:47:24.000000000 +0100
+++ libffi-3.2.1-new/generate-darwin-source-and-headers.py	2020-05-06 10:30:46.000000000 +0200
@@ -194,7 +194,7 @@
         build_target(desktop64_platform, platform_headers)
 
     mkdir_p('darwin_common/include')
-    for header_name, tag_tuples in platform_headers.iteritems():
+    for header_name, tag_tuples in platform_headers.items():
         basename, suffix = os.path.splitext(header_name)
         with open(os.path.join('darwin_common/include', header_name), 'w') as header:
             for tag_tuple in tag_tuples:
