diff -Naur Python-2.7.1.orig/Modules/_ctypes/cfield.c Python-2.7.1/Modules/_ctypes/cfield.c
--- Python-2.7.1.orig/Modules/_ctypes/cfield.c	2015-02-11 13:00:42.000000000 +0100
+++ Python-2.7.1/Modules/_ctypes/cfield.c	2015-02-11 13:01:12.000000000 +0100
@@ -1747,11 +1747,12 @@
 } ffi_type;
 */
 
+#if 0
 #ifdef ffi_type_longdouble
 #undef ffi_type_longdouble
 #endif
   /* This is already defined on OSX */
 ffi_type ffi_type_longdouble = { sizeof(long double), LONGDOUBLE_ALIGN,
                                  FFI_TYPE_LONGDOUBLE };
-
+#endif
 /*---------------- EOF ----------------*/
