--- SDL3-3.2.8/include/build_config/SDL_build_config_ios.h.orig	2025-06-26 21:36:13
+++ SDL3-3.2.8/include/build_config/SDL_build_config_ios.h	2025-06-26 21:37:02
@@ -177,9 +177,9 @@
 
 /* Enable OpenGL ES */
 #if !TARGET_OS_MACCATALYST && !TARGET_OS_VISION
-#define SDL_VIDEO_OPENGL_ES2 1
-#define SDL_VIDEO_OPENGL_ES 1
-#define SDL_VIDEO_RENDER_OGL_ES2 1
+#define SDL_VIDEO_OPENGL_ES2 0
+#define SDL_VIDEO_OPENGL_ES 0
+#define SDL_VIDEO_RENDER_OGL_ES2 0
 #endif
 
 /* Metal supported on 64-bit devices running iOS 8.0 and tvOS 9.0 and newer
