--- configure.in	(revision 18854)
+++ configure.in	(working copy)
@@ -9,7 +9,7 @@
 AC_CANONICAL_HOST
 AC_CANONICAL_TARGET
 
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 AM_INIT_AUTOMAKE(libvorbisidec,1.2.1)
 
@@ -116,15 +116,6 @@
 then
   PKG_CHECK_MODULES(OGG, ogg >= 1.0, HAVE_OGG=yes, HAVE_OGG=no)
 fi
-if test "x$HAVE_OGG" = "xno"
-then
-  dnl fall back to the old school test
-  XIPH_PATH_OGG(, AC_MSG_ERROR(must have Ogg installed!))
-  libs_save=$LIBS
-  LIBS="$OGG_LIBS"
-  AC_CHECK_FUNC(oggpack_writealign, , AC_MSG_ERROR(Ogg >= 1.0 required !))
-  LIBS=$libs_save
-fi
 
 dnl --------------------------------------------------
 dnl Check for library functions
