--- zbarlight-1.2/build/lib.macosx-10.14-x86_64-2.7/zbarlight/__init__.py  2017-03-09 21:43:32.000000000 +0800
+++ zbarlight-1.2/build/lib.macosx-10.14-x86_64-2.7/zbarlight/__init__.py 2018-12-24 12:29:33.000000000 +0800
@@ -2,12 +2,11 @@
 import warnings
 
 from PIL import Image
-import pkg_resources
 
 from ._zbarlight import Symbologies, zbar_code_scanner
 
 
-__version__ = pkg_resources.get_distribution('zbarlight').version
+__version__ = "1.2"
 __ALL__ = ['Symbologies', 'UnknownSymbologieError', 'scan_codes', 'qr_code_scanner']
 
 
 